R/settings_utils.R

```diff
@@ -34,14 +34,21 @@ is_root <- function(path) {
   identical(path, dirname(path))
 }
 
-is_directory <- function(filename) {
-  is_dir <- file.info(filename)$isdir
-
-  !is.na(is_dir) && is_dir
-}
-
-has_config <- function(path, config) {
-  file.exists(file.path(path, config))
+is_directory <- function(filename) isTRUE(file.info(filename)$isdir)
+
+#' Return the first of a vector of files that exists.
+#'
+#' Avoid running 'expensive' [file.exists()] for the full vector,
+#'   since typically the first entries will lead to early exit.
+#' TODO(#2204): check if the implementation should be simpler
+#' @noRd
+first_exists <- function(files) {
+  for (file in files) {
```

Comment:
This can be avoided because `file.exists()` is vectorized.
Not sure if faster, we should check:

```r
file_exists <- file.exists(files)
if (any(file_exists)) {
  files[which.max(file_exists)]
} else {
  NULL
}
```